scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -217,28 +236,26 @@ public void clearJobSignals() throws SignalApiException {
     }
 
     @Override
-    public void waitFor(String signalName) throws SignalApiException {
+    public Map<String, Map<String, String>> waitFor(String signalName) throws SignalApiException {
         init();
+        Map<String, Map<String, String>> signals = new LinkedHashMap<>();
```

Comment:
I am not sure waitFor(signalName) should return a map of map
why not direcly return Map<String, String> ?

Reply:
Map<String, Map<String, String>> represents the signal name and the output values (Map<String, String>). Is it better to return just the output values?

Reply:
yes, as the signal name is already in the parameter, it is known by the api user.

Reply:
ok, I will change it